kong/router/atc.lua

```diff
@@ -96,6 +96,16 @@ local function escape_str(str)
 end
 
 
+local function sanitize_regex(str)
+  -- update named capture opening tag for rust regex::Regex compatibility
+  return str:gsub([[?<]] , [[?P<]])
+            :gsub([[\/]] , [[/]])  -- "\/" => "/"
+            :gsub([[\_]] , [[_]])  -- "\_" => "_"
+            :gsub([[\!]] , [[!]])  -- "\!" => "!"
+            :gsub([[\ ]] , [[ ]])  -- "\ " => " "
```

Comment:
These string substitutions will cause regular expressions like ` ?<` or intended uses of `\\` before of `/`, `_`, `!` or ` ` to be changed unexpectedly.  We should at least cover for these degenerate cases by making the match expressions longer as in. `(?<` and whatever Lua match expression would work for making sure that the backslashes are not preceded by another backslash.

This could still fail to work as designed, leaving users with hard to understand behavior, but it would reduce the likeliness of creating such unexpected behavior.  It may not be very common to use complicated regular expressions that would fail the substitution, but if they are used, it would be very difficult to diagnose what the issue is.